src/main/resources/db/migration/V5__dropping_email_from_session.sql

```diff
@@ -0,0 +1 @@
+ALTER TABLE `user_session` DROP COLUMN `email`;
```

Comment:
This won't work in many cases. If you tried recreating all DB migrations, it will fail. You have your won DB with a dirty state. Clean your DB first, and then apply migrations which need to be done to fix the state:
```
CREATE TABLE IF NOT EXISTS `user_session` (
  `id` BIGINT NOT NULL AUTO_INCREMENT,
  `token` VARCHAR(255) NOT NULL,
  `user_id` BIGINT NOT NULL,
  PRIMARY KEY (`id`),
  UNIQUE (`token`),
  FOREIGN KEY (`user_id`) REFERENCES `user`(`id`)
);
```
It's unacceptable to fix dirty local DB state with Flyway migrations.

Reply:
In this case, are you referring to `IF NOT EXISTS`?

Do you mean that if I have doubts about the existence or partial existence of the table in question, I should drop the table first and the re-add correctly in a later migration?

Reply:
No, the table is defined correctly in the migration file. If you pushed your new migrations to productions, you would destroy the whole state of the DB on production, casing loss of data and probably a long outage. Migrations are not a way to solve your local inconsistencies. Migrations are to manage the target state of the DB and track all actually applied changes.